src/Microsoft.ML.TensorFlow/TensorflowTransform.cs

```diff
@@ -311,26 +319,56 @@ public Mapper(IHostEnvironment env, TensorFlowTransform parent, ISchema inputSch
                 _schema = inputSchema;
                 _inputColIndices = new int[_parent.Inputs.Length];
                 _isInputVector = new bool[_parent.Inputs.Length];
+                _fullySpecifiedShapes = new TFShape[_parent.Inputs.Length];
```

Comment:
>_fullySpecifiedShapes [](start = 16, length = 21)

I feel like this one should be part of Transformer, rather than mapper.
You do estimator.Fit(somedata) and gain transformer, and it resolves it's variable lengths[?,?,3] as [3,3,3].
Not sure it would be right to accept data rather than [3,3,3] to transformer after that.

(Same probably states for _isInputVector, not sure why I didn't put it to Transformer)
@Zruty0 to make sure I correctly understand estimator/transformer business.

Reply:
The problem is that when you instantiate the estimator you only have the TF model and not the IDV, so we don't necessarily know all the dimensions in the shape. However, in order to convert a VBuffer to a Tensor we need to know the fully specified shape.

Instead of having this field, I could instantiate a new shape object on every getter call, using _inputColIndices and _schema to figure out the input size. Do you think this is a good solution?


---
In reply to: [216115594](https://github.com/dotnet/machinelearning/pull/857#discussion_r216115594) [](ancestors = 216115594)

Reply:
I think I just have incorrect understanding of estimator/transformer, and we don't have to have same schema for input fitting and input transforming.

---
In reply to: [216382814](https://github.com/dotnet/machinelearning/pull/857#discussion_r216382814) [](ancestors = 216382814,216115594)